crates/storage/db/src/snapshot/mask.rs

```diff
@@ -0,0 +1,90 @@
+use crate::table::Decompress;
+
+/// Generic Mask helper struct for selecting specific column values to read and decompress.
+///
+/// #### Explanation:
+///
+/// A `NippyJar` snapshot row can contain multiple column values. To specify the column values
+/// to be read, a mask is utilized.
+///
+/// For example, a snapshot with three columns, if the first and last columns are queried, the mask
+/// `0b101` would be passed. To select only the second column, the mask `0b010` would be used.
+///
+/// Since each snapshot has its own column distribution, different wrapper types are necessary. For
+/// instance, `B256` might be the third column in the `Header` segment, while being the second
+/// column in another segment. Hence, `Mask<B256>` would only be applicable to one of these
+/// scenarios.
+///
+/// Alongside, the column selector traits (eg. [`ColumnSelectorOne`]) this provides a structured way
+/// to tie the types to be decoded to the mask necessary to query them.
+#[derive(Debug)]
+pub struct Mask<T, J = (), K = ()>(std::marker::PhantomData<(T, J, K)>);
```

Comment:
Can we rename this to smth that makes it obvious that these are the column types,

like `FIRST`, `SECOND` etc to make that obvious?